core/templates/pages/signup-page/signup-page.component.ts

```diff
@@ -16,213 +16,218 @@
         }
-
-        ctrl.submissionInProcess = true;
-        $http.post(_SIGNUP_DATA_URL, requestParams).then(function(returnValue) {
-          if (returnValue.data.bulk_email_signup_message_should_be_shown) {
-            ctrl.showEmailSignupLink = true;
-            ctrl.submissionInProcess = false;
-            return;
-          }
-          SiteAnalyticsService.registerNewSignupEvent();
-          setTimeout(() => {
-            $window.location.href = decodeURIComponent(
-              UrlService.getUrlParams().return_url);
-          }, SiteAnalyticsService.CAN_SEND_ANALYTICS_EVENTS ? 150 : 0);
-        }, function(rejection) {
-          if (rejection.data && rejection.data.status_code === 401) {
-            ctrl.showRegistrationSessionExpiredModal();
-          }
-          ctrl.submissionInProcess = false;
-        });
-      };
-
-      ctrl.showRegistrationSessionExpiredModal = function() {
-        $uibModal.open({
-          templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
-            '/pages/signup-page/modal-templates/' +
-            'registration-session-expired-modal.template.html'),
-          backdrop: 'static',
-          keyboard: false,
-          controller: 'RegistrationSessionExpiredModalController'
-        }).result.then(function() {}, function() {
-          // Note to developers:
-          // This callback is triggered when the Cancel button is clicked.
-          // No further action is needed.
-        });
-      };
-      ctrl.$onInit = function() {
-        LoaderService.showLoadingScreen('I18N_SIGNUP_LOADING');
-        ctrl.warningI18nCode = '';
-        ctrl.siteName = SITE_NAME;
-        ctrl.submissionInProcess = false;
-        ctrl.usernameCheckIsInProgress = false;
-        ctrl.showEmailSignupLink = false;
-        ctrl.emailSignupLink = BULK_EMAIL_SERVICE_SIGNUP_URL;
-
-        $http.get(_SIGNUP_DATA_URL).then(function(response) {
-          var data = response.data;
-          LoaderService.hideLoadingScreen();
-          ctrl.username = data.username;
-          ctrl.hasEverRegistered = data.has_ever_registered;
-          ctrl.hasAgreedToLatestTerms = data.has_agreed_to_latest_terms;
-          ctrl.showEmailPreferencesForm = data.can_send_emails;
-          ctrl.hasUsername = Boolean(ctrl.username);
-          FocusManagerService.setFocus('usernameInputField');
-        });
-
-        ctrl.blurredAtLeastOnce = false;
-        ctrl.canReceiveEmailUpdates = null;
-      };
-    }
-  ]
-});
+        this.submissionInProcess = false;
+      });
+  }
+
+  showRegistrationSessionExpiredModal(): void {
+    let modalRef = this.ngbModal.open(
+      RegistrationSessionExpiredModalComponent, {
+        backdrop: 'static',
+        keyboard: false
+      });
+
+    modalRef.result.then(() => {}, () => {
+      // Note to developers:
+      // This callback is triggered when the Cancel button is clicked.
+      // No further action is needed.
+    });
```

Comment:
```suggestion
    // Note to developers:
    // This callback is triggered when the Cancel button is clicked.
    // No further action is needed.
    modalRef.result.then(() => {}, () => {});
```

Reply:
Only the second callback is triggered when the cancel button is clicked. So, it makes more sense to place the comment inside the second callback.

Reply:
But the first callback is empty too, should that be explained too?

Reply:
These callbacks are pretty standard and nothing is here that is out of ordinary. The first callback is for confirm modal and second is for cancel modal. In my opinion no comment should be required here. Explaining the api of ng bootstrap should not be our job, everything is there on ng-boostrap's website.

Reply:
Not sure if I agree, the API is confusing, I would assume that if there are no callbacks it should be fine, but we need to pass empty callbacks. If I saw this code for the first time I would like to also see a comment explaining why this is needed.

Reply:
So, what is the reasoning behind not placing the comment inside the callback?

Reply:
added comments for both callbacks